### PR TITLE
Description for Spring Cloud Azure Storage

### PR DESCRIPTION
**Pull request by Matthew-Dong**

### CODE REVIEW

File: spring-cloud-azure-starters/spring-cloud-azure-starter-storage/README.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# Spring Cloud Azure Storage 
+
+The project implements Spring Resource abstraction for Azure Storage which allows you to interact with Azure Storage instances using Spring programming model.
```
Comment (warrenzhu25) [inline spring-cloud-azure-starters/spring-cloud-azure-starter-storage/README.md]: Azure Storage account have fours types. Currently, we only support `blob`. This should be specified.

File: spring-cloud-azure-starters/spring-cloud-azure-starter-storage/README.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+## Feature List 
+
+- [Auto-configuration for Azure Storage](#auto-configuration-for-azure-storage) 
+- [Read a file from Azure Storage](#read-a-file-from-azure-storage)
```
Comment (warrenzhu25) [inline spring-cloud-azure-starters/spring-cloud-azure-starter-storage/README.md]: Read a resource. Resource cloud be a single file or a collection of files. We might support more.

File: spring-cloud-azure-starters/spring-cloud-azure-starter-storage/README.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+- [Write to a file in Azure Storage](#write-to-a-file-in-azure-storage)
```
Comment (warrenzhu25) [inline spring-cloud-azure-starters/spring-cloud-azure-starter-storage/README.md]: Same as above

File: spring-cloud-azure-starters/spring-cloud-azure-starter-storage/README.md (hunk 2 above)
Comment (warrenzhu25) [reply spring-cloud-azure-starters/spring-cloud-azure-starter-storage/README.md]: For Azure Storage blob.

File: spring-cloud-azure-starters/spring-cloud-azure-starter-storage/README.md (hunk 1 above)
Comment (warrenzhu25) [reply spring-cloud-azure-starters/spring-cloud-azure-starter-storage/README.md]: Maybe a link to azure storage blob here.

File: spring-cloud-azure-starters/spring-cloud-azure-starter-storage/README.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+```
+
+### Read a file from Azure Storage
+You can use the annotation of `@Value("blob://{containerName}/{blobName}")` to map a `Resource` with a file in [Azure Blob Storage](https://azure.microsoft.com/en-us/services/storage/blobs/). The following code snippet shows how to read a file with `getInputStream()` method.
+
+```
+@Value("blob://{containerName}/{blobName}")
+private Resource blobFile;
+
+@RequestMapping(value = "/", method = RequestMethod.GET)
```
Comment (warrenzhu25) [inline spring-cloud-azure-starters/spring-cloud-azure-starter-storage/README.md]: Should not put REST method here.

**Comment by Matthew-Dong** [comment]
Updated the tutorial about:

- the scope of Azure Blob storage 
- read and write to a resource (instead of a file)
- add the table of application properties 
- other wording changes

File: spring-cloud-azure-starters/spring-cloud-azure-starter-storage/README.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+If you are using Gradle, add the following dependency to your project. 
+```
+dependencies {
+    compile group: 'com.microsoft.azure', name: 'spring-cloud-azure-starter-storage'
+}
+```
+
+The [`spring-cloud-azure-starter-storage`](spring-cloud-azure-starters/spring-cloud-azure-starter-storage/) provides the following configuration options in `application.properties`.
+
+Name | Description | Required | Default
```
Comment (warrenzhu25) [inline spring-cloud-azure-starters/spring-cloud-azure-starter-storage/README.md]: Add property `spring.cloud.azure.storage.enabled`, which has default value `true` and is not required. This gives user the ability to disable auto config

**Comment by Matthew-Dong** [comment]
Update as follows: 

- Add config option for turning on and off Spring Cloud Azure Storage 
- Wording changes

**Pull request merged**